webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +95,1093 @@ public WebSiteSessionController createComponentSessionController(
+    String nomSite = request.getParameter(SITE_NAME_PARAM);
+    String description = request.getParameter(DESCRIPTION_PARAM);
+    String nomPage = request.getParameter(PAGE_NAME_PARAM);
+    String tempPopup = request.getParameter(POPUP_PARAM);
+    String letat = request.getParameter("etat");
+    String listeIcones = request.getParameter(ICON_LIST_PARAM);
+    String listeTopics = request.getParameter(TOPIC_LIST_PARAM);
+
+    int popup = (tempPopup != null) && (tempPopup.length() > 0) ? 1 : 0;
+    int etat = StringUtil.isDefined(letat) ? Integer.parseInt(letat) : -1;
+
+    List<String> listIcons = new ArrayList<>();
+    for (String icon : listeIcones.split(",")) {
+      if (!icon.isEmpty()) {
+        listIcons.add(icon);
+      }
+    }
+    SiteDetail ancien = scc.getWebSite(id);
+    int type = ancien.getType();
+          /* update description en BD */
+    SiteDetail descriptionSite2 =
+        new SiteDetail(id, scc.getComponentId(), nomSite, description, nomPage, type, null, null,
+            etat, popup);
 
-          if (publish) {
-            arrayToClassify.add(id);
-            scc.publish(arrayToClassify); // set etat du site a 1
-          }
-        } else if (action != null && action.equals("deleteWebSites")) {
+    scc.updateWebSite(descriptionSite2);
+    if (listIcons.size() > 0) {
+      scc.associateIcons(id, listIcons);
+    }
 
-          ArrayList<String> listToDelete = new ArrayList<>();
+          /* publications : classer le site dans les themes cochés */
+    ArrayList<String> arrayToClassify = new ArrayList<>();
+    boolean publish = false;
+    ArrayList<String> arrayTopic = new ArrayList<>();
+    for (String idTopic : listeTopics.split(",")) {
+      if (!idTopic.isEmpty()) {
+        arrayTopic.add(idTopic);
+        publish = true;
+      }
+    }
 
-          String liste = request.getParameter("SiteList");
+    scc.updateClassification(id, arrayTopic);
 
-          int begin = 0;
-          int end = 0;
-          end = liste.indexOf(',', begin);
-          String idToDelete;
-          SiteDetail info;
-          int type;
+    arrayToClassify.add(id);
+    if (publish) {
+      scc.publish(arrayToClassify); // set etat du site a 1
+    } else {
+      scc.dePublish(arrayToClassify);
+    }
+  }
 
-          while (end != -1) {
-            idToDelete = liste.substring(begin, end);
-            listToDelete.add(idToDelete);
+  private void deleteSite(final WebSiteSessionController scc, final HttpRequest request)
+      throws WebSitesException {
+    ArrayList<String> listToDelete = new ArrayList<>();
+    String liste = request.getParameter("SiteList");
+
+    for (String siteId : liste.split(",")) {
+      if (!siteId.isEmpty()) {
+        listToDelete.add(siteId);
+
+        // recup info sur ce webSite
+        SiteDetail info = scc.getWebSite(siteId);
+        // type = 0 : site cree, type = 1 : site bookmark, type = 2 : site upload
+        if (info.getType() != 1) { // type != bookmark
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ATrailingCommentCheck)